packages/core-api/src/apis/ApiProvider.test.tsx

```diff
@@ -51,6 +51,35 @@ describe('ApiProvider', () => {
     renderedHoc.getByText('hoc message: hello');
   });
 
+  it('should provide nested access to apis', () => {
+    const aRef = createApiRef<string>({ id: 'a', description: '' });
+    const bRef = createApiRef<string>({ id: 'b', description: '' });
+
+    const MyComponent = () => {
+      const a = useApi(aRef);
+      const b = useApi(bRef);
+      return (
+        <div>
+          a={a} b={b}
+        </div>
+      );
+    };
+
+    const renderedHook = render(
+      <ApiProvider
+        apis={ApiRegistry.from([
+          [aRef, 'x'],
+          [bRef, 'y'],
+        ])}
+      >
+        <ApiProvider apis={ApiRegistry.from([[aRef, 'z']])}>
+          <MyComponent />
+        </ApiProvider>
+      </ApiProvider>,
+    );
+    renderedHook.getByText('a=z b=y');
```

Comment:
I know it's not needed, but is it still good practice to have an assert here?

Reply:
I do it when I haven't yet disabled the lint rule and can't be arsed xP

I dunno, what'd be the benefit of doing it? So far I haven't hit any trouble because of omitting it, but could be convinced :grin:

Reply:
I mean I could be convinced that it's not a good idea too.. haha.

I just think it reads a little strange. but so does:

```ts
expect(() => renderedHook.getByText(..)).not.toThrow()
```

Maybe something like:

```ts
expect(await renderedHook.getByText('...')).toBeInTheDocument()
```

is nicer?

Reply:
or omit the await if it's not needed

Reply:
just conistency I guess, as other tests to the latter.

Reply:
yeah `expect(renderedHook.getByText('...')).toBeInTheDocument()` would be the alternative, but still cba using that everywhere :p